pass.el

```diff
@@ -342,6 +345,54 @@ indented according to INDENT-LEVEL."
           (forward-line -1)
           (pass-closest-entry)))))
 
+(defun pass-otp-options (option)
+  "Dispatch otp actions depending on user OPTION input.
+Display help message with OTP functionality options."
+  (interactive
+   (list
+    (read-char-choice "[i] Insert, [a] Append, [s] Append from screenshot, [o] Copy token, [u] Copy URI, or [C-g] to abort: "
+                      '(?i ?a ?s ?o ?u))))
+  (unless (require 'password-store-otp nil t)
+    (user-error "You cannot do this without installing `password-store-otp' first"))
+  (pcase option
+    (?i (pass-otp-insert))
+    (?a (pass-otp-append))
+    (?s (pass-otp-from-screenshot))
+    (?o (pass-otp-token-copy))
+    (?u (pass-otp-uri-copy))))
+
+(defun pass-otp-token-copy ()
+  "Add OTP Token from closest entry to kill ring."
+  (interactive)
+  (pass--with-closest-entry entry
+    (password-store-otp-token-copy entry)))
+
+(defun pass-otp-uri-copy ()
+  "Add OTP URI from closest entry to kill ring."
+  (interactive)
+  (pass--with-closest-entry entry
+    (password-store-otp-uri-copy entry)))
+
+(defun pass-otp-insert ()
+  "Insert an OTP URI entry to the password-store.
+The password is read from user input."
+  (interactive)
+  (call-interactively #'password-store-otp-insert)
```

Comment:
is it possible to use `pass--with-closest-entry` here to avoid typing the entry name at point?

Reply:
Yes, I dislike this too.

I left it this way because of how `password-store-otp-insert` (and `-append` and `-append-from-image`) works. I want those functions to be interactive (to mirror password-store's `insert`, `edit`, `remove`, etc...) and when I called them here "interactively" I don't have a way to specify the first argument `entry` of the interaction, at least according to what I understood from [`call-interactively`'s documentation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Interactive-Call.html).

Another solution might be to use something like the internal `password-store-otp--insert`. Before going this way, I'd like to hear your thoughts on volrath/password-store-otp.el#2 so that I can remove the second hyphen in that function, possibly change the name, and push it all as a new version.

Reply:
Oh, adding a `url` parameter to `pass-otp-insert` and `pass-otp-append` might also work, but I didn't see that pattern almost elsewhere in `pass.el`, so I'm not sure if it'd be the right way to go.

Reply:
@NicolasPetton: What you did seems a bit non-standard I think. Why didn't you write something like the following for your interactive commands?

```emacs
(defun pass-view (entry)
  "Visit the ENTRY.
Interactively, view entry at point."
  (interactive (list (pass-closest-entry))
  (find-file (concat (f-join (password-store-dir) entry) ".gpg")))
```

@volrath:

> adding a url parameter to pass-otp-insert and pass-otp-append might also work

indeed, but that's a bit weird with current code.

What about calling non interactively instead:

```emacs
(defun pass-otp-insert ()
  "Insert an OTP URI entry to the password-store.
The password is read from user input."
  (interactive)
  (password-store-otp-insert (pass-closest-entry) (read-passwd "OTP URI: " t))
  (pass-update-buffer))
```

Even though that is duplicating the `read-passwd` logic from `password-store-otp-insert` (bad), I think this is reasonably ok as `pass` should be responsible for user interactions while `password-store-otp` is used as a library.

Reply:
@DamienCassou Done, eab8de0. I implemented the "closest entry" thing in both `pass-otp-append` and `pass-otp-from-screenshot`, but not on `pass-otp-insert`, because for insert it makes sense to ask for the entry, which will be a new entry. This follows the `pass otp insert` command.

Reply:
@DamienCassou @NicolasPetton, about Damien's inquiry of having `pass-closest-entry` be a function instead of a macro, I think it's a good idea. We could also make it something like `(pass-closest-entry-or-prompt)`, which would try to find a closest entry or prompt the user to specify one if there's no entry at point. This way, things like killing, editing, appending, renaming and so on, gain a new level of accessibility.

I think this should be implemented in a different PR though.

Reply:
@DamienCassou Now I don't remember why I did it this way :)

Reply:
@volrath I agree, let's do that in a different PR>